conjure-java-jaxrs-client/src/main/java/com/palantir/conjure/java/client/jaxrs/DialogueFeignClient.java

```diff
@@ -57,18 +57,23 @@
 import java.util.List;
 import java.util.Locale;
 import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import org.immutables.value.Value;
 
 /**
  * {@link DialogueFeignClient} is an adapter from {@link feign.Client} to {@link Channel Dialogue Channel}
  * taking advantage of the superior observability and stability provided by Dialogue.
  */
 final class DialogueFeignClient implements feign.Client {
 
+    private static final String REQUEST_URL_PATH_PARAM = "request-url";
     private static final String PATH_TEMPLATE = "hr-path-template";
     private static final Splitter pathSplitter = Splitter.on('/');
     private static final Splitter querySplitter = Splitter.on('&').omitEmptyStrings();
     private static final Splitter queryValueSplitter = Splitter.on('=');
 
+    private final ConcurrentHashMap<FeignEndpointKey, EndpointChannel> endpointChannelCache = new ConcurrentHashMap<>();
```

Comment:
Using a `ConcurrentHashMap` instead of a Caffeine cache because I don't think we have to worry about the fact that this is unbounded. The number of entires here will be bounded by the fact that this client comes from some JAX-RS interface.